src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -528,42 +528,67 @@ internal static Matrix3x2 GetTransform(UIElement from, UIElement to)
 				return Matrix3x2.Identity;
 			}
 
-#if UNO_REFERENCE_API // Depth is defined properly only on WASM and Skia
-			// If possible we try to navigate the tree upward so we have a greater chance
-			// to find an element in the parent hierarchy of the other element.
-			if (to is { } && from.Depth < to.Depth)
+#if __SKIA__
+			if (from.Visual?.TotalMatrix is { } fromMatrix && (to is null || to.Visual?.TotalMatrix is { }))
```

Comment:
I'm not really sure about this. TotalMatrix could be non-null but having an outdated value (when we're in between arrange and render)

Reply:
Maybe we should set `TotalMatrix` when measure or arrange is invalidated? I'm not sure though if there are any edge cases where `TotalMatrix` could change without arrange being invalidated?

Reply:
> I'm waiting for your lifecycle stuff to merge to see what we can do about that.

Yeah it also *might* be possible to do something with the enhanced lifecycle by controlling when we do render ourselves. But I'm not 100% sure if that will end up being possible/viable.